### ISSUE TITLE
Merge IndexManager and CacheManager 

### ISSUE THREAD

**Issue by gfl94** [InitialDLab/Simba#15]
We can integrate IndexManger together with CacheManager,  IndexManager use a bytebuffer for IndexInfo storage, and CacheManager use a  column store for CacheData and CacheInfo, perhaps we can put them together?

**Comment by Skyprophet** [comment]
No, this is not the case. I mean we can give users a choice to replace the `Array[InternalRow]` in `IPartition` to columnar format like the original Spark SQL cache mechanism.
